iceoryx2-bb/lock-free/src/mpmc/bit_set.rs

```diff
@@ -198,13 +212,48 @@ pub mod details {
                     Ordering::Relaxed,
                     Ordering::Relaxed,
                 ) {
-                    Ok(_) => return true,
+                    Ok(_) => {
+                        self.active_bits.fetch_add(1, Ordering::Relaxed);
+                        return true;
+                    }
+                    Err(v) => current = v,
+                }
+            }
+        }
+
+        fn clear_bit(&self, id: usize) -> bool {
+            let index = id / BITSET_ELEMENT_BITSIZE;
+            let bit = id % BITSET_ELEMENT_BITSIZE;
+
+            let data_ref = unsafe { &(*self.data_ptr.as_ptr().add(index)) };
+            let mut current = data_ref.load(Ordering::Relaxed);
+            let bit = 1 << bit;
+
+            loop {
+                if current & bit == 0 {
+                    return false;
+                }
+
+                let current_with_bit = current & !bit;
+
+                match data_ref.compare_exchange(
+                    current,
+                    current_with_bit,
+                    Ordering::Relaxed,
+                    Ordering::Relaxed,
+                ) {
+                    Ok(_) => {
+                        self.active_bits.fetch_sub(1, Ordering::Relaxed);
+                        return true;
+                    }
```

Comment:
It is not a problem.

1. It always keeps the container in a consistent and consistent state.
2. When a bit was set, but the active_bits counter was not incremented, it is possible that:
  2.1 The current bit is not returned in `reset_next`. This is no problem from the user's point of view since the set operation is not yet finished since the function has not yet returned.
  2.2 The current bit is returned, but an older bit is not returned since the counter still needs to be set. But as soon as this function returns, it can be returned.

But! When an application crashes in 2. the whole thing becomes inconsistent :( So we have to remove the active_bits.